10_datastructures_lab/01.py

```diff
@@ -1,10 +1,19 @@
-"""
-Write a program that takes two strings
-from the user and checks if they represent
-a valid user name.
-Valid users and passwords:
-    apple => red
-    lettuce => green
-    lemon => yellow
-    orange => orange
-"""
+import os
+import sys
+program_name = sys.argv[0]
+hosts = {}
+
+with open("./hosts.txt", 'r') as fin:
+    for line in fin:
+        item = line.split("=")
+        hosts[item[0]] = item[1]
+from sys import argv
+for hostname in argv:
+    if hosts.has_key(hostname):
+        print hosts[hostname]
+    elif hostname == program_name:
+        continue
+    else:
+        print "the hostname you entered is invalid"
+    
+

```

Comment:
A couple of issues:
* There is no need for ./  , it is clearer to write ```open('hosts.txt', 'r')```
* Lines 8-9 are better written like this:

```
        name, ip = line.split("=")
       hosts[name] = ip
```
* Instead of ```hosts.has_key(hostname)``` it is nicer to write ```hostname in hosts```
* You could iterate over ```for hostname in argv[1:]``` and then there is no need for the 14-15 lines which are confusing anyway.